test/unit/test_h5_loader.py

```diff
@@ -129,5 +146,51 @@ def test_get_data():
     ]
     expected = [(44, 59, 41), [255.0, 0.0, 68.359276, 65.84009]]
     loader.close()
-    assert got[0] == expected[0]
-    assert check_equal(np.array(got[1]), np.array(expected[1])).all()
+    if got[0] == expected[0]:
+        assert check_equal(np.array(got[1]), np.array(expected[1])).all()
+    else:
+        raise AssertionError
+
+
+def test_init_incompatible_conditions():
+    """
+    check if the initialisation works as expected and raises an error when
+    directories to ungrouped files is given but grouped variable is set to
+    True
+    """
+    dir_path = "data/test/h5/paired/test"
+    name = "fixed_images"
+    with pytest.raises(IndexError) as execinfo:
+        H5FileLoader(dir_path=dir_path, name=name, grouped=True)
+    msg = " ".join(execinfo.value.args[0].split())
+    assert "index out of range" in msg
+
+
+def test_get_data_incompatible_args():
+    """
+    check if the get_data method works as expected and raises an error when
+    data is ungrouped but index is not an int
+    """
+    dir_path = "data/test/h5/paired/test"
```

Comment:
Change to ./data/test...? relative file path works locally for me vs the current solution. Will it pass travis?